native/src/main/java/io/ballerina/stdlib/etl/nativeimpl/EtlCategorization.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ * 
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.etl.nativeimpl;
+
+import io.ballerina.runtime.api.Environment;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.utils.TypeUtils;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BRegexpValue;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.api.values.BTypedesc;
+import io.ballerina.stdlib.etl.utils.ErrorUtils;
+import org.ballerinalang.langlib.regexp.Matches;
+
+import static io.ballerina.stdlib.etl.utils.CommonUtils.convertJSONToNestedBArray;
+import static io.ballerina.stdlib.etl.utils.CommonUtils.getFieldType;
+import static io.ballerina.stdlib.etl.utils.CommonUtils.initializeNestedBArray;
+import static io.ballerina.stdlib.etl.utils.CommonUtils.isFieldExist;
+import static io.ballerina.stdlib.etl.utils.Constants.CATEGORIZE_SEMANTIC;
+import static io.ballerina.stdlib.etl.utils.Constants.CLIENT_CONNECTOR_ERROR;
+import static io.ballerina.stdlib.etl.utils.Constants.CLIENT_REQUEST_ERROR;
+import static io.ballerina.stdlib.etl.utils.Constants.FLOAT;
+import static io.ballerina.stdlib.etl.utils.Constants.IDLE_TIMEOUT_ERROR;
+import static io.ballerina.stdlib.etl.utils.Constants.INT;
+import static io.ballerina.stdlib.etl.utils.Constants.INT_OR_FLOAT;
+import static io.ballerina.stdlib.etl.utils.Constants.STRING;
+
+/**
+ * This class hold Java external functions for ETL - data categorization APIs.
+ *
+ * * @since 0.8.0
+ */
+@SuppressWarnings("unchecked")
+public class EtlCategorization {
+
+    public static Object categorizeNumeric(BArray dataset, BString fieldName, BArray rangeArray, BTypedesc returnType) {
+        if (!isFieldExist(dataset, fieldName)) {
+            return ErrorUtils.createFieldNotFoundError(fieldName);
+        }
+        String fieldType = getFieldType(returnType, fieldName);
+        if (!fieldType.contains(INT) && !fieldType.contains(FLOAT)) {
+            return ErrorUtils.createInvalidFieldTypeError(fieldName, INT_OR_FLOAT, fieldType);
+        }
+        float lowerBound = Float.parseFloat(String.valueOf(rangeArray.get(0)));
```

Comment:
Why do we have to get the `String.valueOf` for the fields? Can't we directly obtain the `float` value from the field?